proto/backup.proto

```diff
@@ -185,8 +187,27 @@ message GCS {
     // You can get one from https://console.cloud.google.com/apis/credentials, and
     // copy the content, set it as string here.
     string credentials_blob = 6;
+    Bucket bucket_info = 7;
 }
 
+// S3 and GCS can be transitioned to use this
+message Bucket {
+    string endpoint = 1; // use a different endpoint (host + url)
+    string host = 2; // use a different host
+    string region = 3;
+    string bucket = 4;
+    string prefix = 5;
+    string storage_class = 6;
```

Comment:
i think `storage_class` belongs to the `attrs` field.

Reply:
If anything is expected to be common between clouds I put it in Bucket. Both AWS and GCP support specifying a storage class so I expect everyone else to as well. Maybe it is not available in many clouds?

Reply:
I just think that endpoint/host, region, bucket, prefix are all essential for locating the resource, so having storage_class here a bit strange 😅.

Reply:
Maybe I would call your concept a BucketLocation. I would much prefer to not use attrs for commonly used fields. attrs can have typos and are more difficult to document.